### PR TITLE
Suport hard delete in core(database and ES)

### PR DESCRIPTION
**Pull request by chopadetejaswini**
Support hard delete in core

### CODE REVIEW

File: java/registry/src/main/java/dev/sunbirdrc/registry/controller/FileStorageController.java [hunk 1]
```diff
@@ -40,12 +45,31 @@ public ResponseEntity<DocumentsResponse> save(@RequestParam MultipartFile[] file
...
+    public ResponseEntity<DocumentsResponse> update(@RequestParam MultipartFile[] files,
+                                                   @PathVariable String entity,
                                                     @PathVariable String entityId,
-                                                    @PathVariable String property,
-                                                    @RequestBody List<String> files,
-                                                    HttpServletRequest httpServletRequest) {
+                                                   @PathVariable String documentId,
+                                                   @PathVariable String property,
+                                                    HttpServletRequest httpServletRequest) throws ServerException, InsufficientDataException, ErrorResponseException, IOException, NoSuchAlgorithmException, InvalidKeyException, InvalidResponseException, XmlParserException, InternalException//MultipartFile[] represents the updated file(s) that will be uploaded to the server.
```
Comment (tejash-jl) [inline java/registry/src/main/java/dev/sunbirdrc/registry/controller/FileStorageController.java:54]: don't throw any exception from controller,
handle all the error inside

File: java/registry/src/main/java/dev/sunbirdrc/registry/controller/FileStorageController.java [hunk 2]
```diff
@@ -40,12 +45,31 @@ public ResponseEntity<DocumentsResponse> save(@RequestParam MultipartFile[] file
...
+    //httpServletRequest extract information about the incoming request
+    {
+        try {
+            registryHelper.authorize(entity, entityId, httpServletRequest);// authorize() to authorize the request
```
Comment (tejash-jl) [inline java/registry/src/main/java/dev/sunbirdrc/registry/controller/FileStorageController.java:58]: remove all comments like this,
https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/

File: java/registry/src/main/java/dev/sunbirdrc/registry/service/FileStorageService.java [hunk 3]
```diff
@@ -49,6 +48,16 @@ public void save(InputStream inputStream, String objectName) throws Exception {
 		logger.info("File has successfully saved");
 	}
 
+
+	public void update(InputStream inputStream, String objectName) throws Exception//inputStream is a file to be saved,Returns a new InputStream that reads no bytes. The returned stream is initially open. The stream is closed by calling the close() method. Subsequent calls to close() have no effect.
```
Comment (tejash-jl) [inline java/registry/src/main/java/dev/sunbirdrc/registry/service/FileStorageService.java:52]: where is the func used?

File: java/registry/src/main/java/dev/sunbirdrc/registry/service/FileStorageService.java [hunk 4]
```diff
@@ -139,4 +150,46 @@ public ComponentHealthInfo getHealthInfo() {
...
+						StatObjectArgs.builder().bucket(bucketName).object(objectName).build()
+				) != null;
+
+				if (objectExists) {
+					save(file.getInputStream(),objectName);
+					documentsResponse.addDocumentLocation(objectName);
+					return documentsResponse;
+				}
+
+				for (MultipartFile file1 : files) {
```
Comment (varadeth) [inline java/registry/src/main/java/dev/sunbirdrc/registry/service/FileStorageService.java:173]: I would say lets not save the file if it is not present. Lets only update the file if present or else throw appropriate error

**Review by varadeth — CHANGES_REQUESTED** [review]

**Pull request closed without merging**